application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelStatus.java

```diff
@@ -0,0 +1,205 @@
+ */
+public final class ChannelStatus {
+    public static final boolean FREE = false;
+    public static final boolean BUSY = true;
+
+    private static final String FREE_STATUS = ":green_circle:";
+    private static final String BUSY_STATUS = ":red_circle:";
+
+    private final long channelId;
+    private long userId;
+    private volatile boolean isBusy;
+    private String name;
+
+    /**
+     * Creates an instance of a Channel Status.
+     * <p>
+     * This does not validate the id as that requires the JDA. Any ChannelStatus that gets created
+     * with an invalid id *should* be ignored and won't be invoked. (Since the Channel statuses are
+     * selected by retrieval of channels id's via the guild id, before retrieving the relevant
+     * ChannelStatuses).
+     * 
+     * @param id the long id of the {@link net.dv8tion.jda.api.entities.TextChannel} to monitor.
+     */
+    ChannelStatus(final long id) {
+        channelId = id;
+        isBusy = BUSY;
+        name = Long.toString(id);
+    }
+
+    /**
+     * Retrieves whether the channel is currently busy/free.
+     * <p>
+     * This value is volatile but is not thread safe in any other way. While statuses change
+     * frequently, each individual status instance *should* only be modified from a single source,
+     * since it represents only a single channel and modification will only be triggered by activity
+     * in that one channel.
+     * 
+     * @return the current stored status related to the channel id.
+     */
+    public boolean isBusy() {
+        return isBusy;
+    }
+
+    /**
+     * Method to test if an id is the same as the id of the help requester who most recently posted
+     * a question.
+     *
+     * @param userId the id to test
+     * @return {@code true} if the id value passed in is the same as the value of the user who most
+     *         recently changed the status to 'busy'. {@code false} otherwise.
+     */
+    public boolean isAsker(final long userId) {
+        return this.userId == userId;
+    }
+
+    /**
+     * Retrieves the id for the {@link net.dv8tion.jda.api.entities.TextChannel} that this instance
+     * represents. There is no guarantee that the id is valid according to the {@link JDA}.
+     * 
+     * @return the {@link net.dv8tion.jda.api.entities.TextChannel} id.
+     */
+    public long getChannelId() {
+        return channelId;
+    }
+
+    /**
+     * Retrieves the locally stored name of the {@link net.dv8tion.jda.api.entities.TextChannel}
+     * this represents. This value is initialised to the channel id and as such is never null. The
+     * name should first be set by retrieving the name the {@link JDA} currently uses, before
+     * calling this.
+     * <p>
+     * The recommended value to use is {@link TextChannel#getAsMention()}.
+     *
+     * @return The currently stored name of the channel, originally the long id as string.
```

Comment:
NIT: "originally the long id as string". is this really implementation detail you want to expose via doc? means you cant change it anymore in the future since people might rely on it